core/templates/domain/objects/NumberWithUnitsObjectFactory.ts

```diff
@@ -94,8 +94,16 @@ export class NumberWithUnits {
       unitsString);
 
     if (this.type === 'real') {
+      // If numerator and whole number is zero, then we can say that there is
+      // no fraction part.
+      if (this.fraction.numerator !== 0 && this.fraction.wholeNumber !== 0) {
```

Comment:
The if-statement doesn't quite follow the logic mentioned in the comment. I think it should be an or-condition instead of an and-condition.
Also, shouldn't just a non-zero check on the numerator be enough?

Reply:
Correct, will make it an or-condition.

But, I think we should skip this check, as it would break [this](https://github.com/oppia/oppia/blob/14d9f1e8b59fd6dd5c3ccea8fe4fb903379e20f9/extensions/interactions/NumberWithUnits/directives/number-with-units-validation.service.ts#L84-L93) code to check equivalency. What do you think so?
(cc: @seanlip )

Reply:
I'm not that familiar with this part of the code -- I think this is @iamprayush's and @aks681's call.

Reply:
Hi @iamprayush -- can you take a look? Thanks!

Reply:
Hi, sorry, I'll leave it to @iamprayush on the actual check, since I'm also not familiar with this interaction's checks and how the 2 snippets relate. Maybe @kevintab95 can help?